src/textual/_border.py

```diff
@@ -261,8 +274,70 @@ def get_box(
     )
 
 
+@lru_cache(maxsize=1024)
+def render_border_label(
+    label: str,
+    is_title: bool,
+    name: EdgeType,
+    width: int,
+    inner_style: Style,
+    outer_style: Style,
+    style: Style,
+) -> Segment:
```

Comment:
This can't be right. If the title contains markup,  like "[bold]Hello [magenta]World[/][blink]![/]", then the it will consist of several segments. But you are returning a single segment.

Reply:
You were right, I misunderstood how text rendering works.